include/stdx/rollover.hpp

```diff
@@ -0,0 +1,155 @@
+#pragma once
+
+#include <limits>
+#include <type_traits>
+
+#if __has_include(<compare>)
+#include <compare>
+#endif
+
+namespace stdx {
+inline namespace v1 {
+
+template <typename T> struct rollover_t {
+    using underlying_t = T;
```

Comment:
issue: there should be a `static_assert` to ensure only a compatible integral type is used.

Reply:
Yes, I intended that but forgot!